CoreQualityControlView/test/unit/src/au/gov/asd/tac/constellation/views/qualitycontrol/daemon/QualityControlAutoVetterNGTest.java

```diff
@@ -74,4 +86,219 @@ public void testGetInstance() {
+        assertFalse(listener.getStateChangedStatus());
+    }
+
+    // Test if multiple buttonlisteners get fired correctly within the update state.
+    @Test
+    public void testUpdateQualityControlState() throws InterruptedException {
+        graph = null;
+
+        // add observer1 of the button state
+        QualityControlAutoVetter.destroyInstance();
+        final testObserver observer1 = new testObserver();
+        QualityControlAutoVetter.getInstance().addObserver(observer1);
+
+        // add observer1 of the button state
+        final testObserver observer2 = new testObserver();
+        QualityControlAutoVetter.getInstance().addObserver(observer2);
+
+        // Check initial status
+        assertFalse(observer1.getCanRunStatus());
+        assertFalse(observer2.getCanRunStatus());
+
+        QualityControlAutoVetter.updateQualityControlState(graph);
+
+        // Sleep until after pluginExecution thread has returned
+        Thread.sleep(1000);
+
+        // Check updated status
+        assertTrue(observer1.getCanRunStatus());
+        assertTrue(observer2.getCanRunStatus());
+    }
+
+    @Test
+    public void testGraphChangedNoGraph() throws InterruptedException {
+        QualityControlAutoVetter.destroyInstance();
+        QualityControlAutoVetter instance = QualityControlAutoVetter.getInstance();
+
+        assertNull(instance.getCurrentGraph());
+        assertEquals(instance.getlastGlobalModCount(), (long) 0);
+        assertEquals(instance.getlastCameraModCount(), (long) 0);
+
+        QualityControlAutoVetter.getInstance().graphChanged(null);
+
+        // check that no attribute mod counts have unnecesarily changed.
+        assertNull(instance.getCurrentGraph());
+        assertEquals(instance.getlastGlobalModCount(), (long) 0);
+        assertEquals(instance.getlastCameraModCount(), (long) 0);
+    }
+
+    // Test commented out as TopComponent launches GUI panels which don't execute in the test environment
+    @Test
+    public void testGraphChangedWithGraph() throws InterruptedException {
+        QualityControlAutoVetter.destroyInstance();
+        QualityControlAutoVetter instance = QualityControlAutoVetter.getInstance();
+
+        assertNull(instance.getCurrentGraph());
+        assertEquals(instance.getlastGlobalModCount(), (long) 0);
+
+        // Open a new graph
+        graph = new DualGraph(SchemaFactoryUtilities.getSchemaFactory(VisualSchemaFactory.VISUAL_SCHEMA_ID).createSchema());
+
+        // Add camera attribute
+        final WritableGraph wg = graph.getWritableGraph("TEST", true);
+        VisualConcept.GraphAttribute.CAMERA.ensure(wg);
+        wg.commit();
+
+        // Change camera attribute
+        final WritableGraph wg1 = graph.getWritableGraph("TEST", true);
+        final Camera camera = new Camera();
+        camera.setVisibilityLow(0.67f);
+        wg1.setObjectValue(VisualConcept.GraphAttribute.CAMERA.get(wg1), 0, camera);
```

Comment:
you should be able ensure and set the attribute value with the same write lock. I would also put any changes to the writable graph in a try-finally if possible